src/core/ext/filters/client_channel/http_proxy.c

```diff
@@ -80,6 +81,39 @@ static bool proxy_mapper_map_name(grpc_exec_ctx* exec_ctx,
     grpc_uri_destroy(uri);
     return false;
   }
+  char* no_proxy_str = gpr_getenv("no_proxy");
+  if (no_proxy_str != NULL) {
+    char *server_host;
+    char *server_port;
+    if (!gpr_split_host_port(uri->path[0] == '/' ? uri->path + 1 : uri->path, &server_host, &server_port)) {
+      gpr_log(GPR_INFO, "unable to split host and port, not checking no_proxy list for host '%s'", server_uri);
+    } else {
+      size_t uri_len = strlen(server_host);
+      size_t i = 0;
+      size_t marker = 0;
+      while (no_proxy_str[i] != 0) {
```

Comment:
I suggest using `gpr_string_split()` to split up the entries in the environment variable value.  Then we can just iterate through the array, comparing `server_host` against each element in turn.

https://github.com/grpc/grpc/blob/master/src/core/lib/support/string.h#L84